exporter/prometheusremotewriteexporter/helper.go

```diff
@@ -340,36 +340,12 @@ func addSingleIntHistogramDataPoint(pt *otlp.IntHistogramDataPoint, metric *otlp
 	// treat count as a sample in an individual TimeSeries
 	count := &prompb.Sample{
 		Value:     float64(pt.GetCount()),
-		Timestamp: time,
+		Timestamp: timeStamp,
 	}
 	countlabels := createLabelSet(pt.GetLabels(), externalLabels, nameStr, baseName+countStr)
 	addSample(tsMap, count, countlabels, metric)
 
-	// cumulative count for conversion to cumulative histogram
-	var cumulativeCount uint64
-
-	// process each bound, ignore extra bucket values
-	for index, bound := range pt.GetExplicitBounds() {
-		if index >= len(pt.GetBucketCounts()) {
-			break
-		}
-		cumulativeCount += pt.GetBucketCounts()[index]
-		bucket := &prompb.Sample{
-			Value:     float64(cumulativeCount),
-			Timestamp: time,
-		}
-		boundStr := strconv.FormatFloat(bound, 'f', -1, 64)
-		labels := createLabelSet(pt.GetLabels(), externalLabels, nameStr, baseName+bucketStr, leStr, boundStr)
-		addSample(tsMap, bucket, labels, metric)
-	}
-	// add le=+Inf bucket
-	cumulativeCount += pt.GetBucketCounts()[len(pt.GetBucketCounts())-1]
-	infBucket := &prompb.Sample{
-		Value:     float64(cumulativeCount),
-		Timestamp: time,
-	}
-	infLabels := createLabelSet(pt.GetLabels(), externalLabels, nameStr, baseName+bucketStr, leStr, pInfStr)
-	addSample(tsMap, infBucket, infLabels, metric)
+	toHistogramSamples(timeStamp, baseName, pt.GetLabels(), externalLabels, metric, pt.GetExplicitBounds(), pt.GetBucketCounts(), tsMap)
```

Comment:
This has a lot of parameters and is difficult to read and understand. Can we pass pt and reduce the count of parameters by 2? We could also avoid passing timeStamp and recalculate it since it is a trivial calculation.